File: Searching/binary_search.py
def binary_search(list1,item):
    first = 0
    last = len(list1)-1
    f=0
    a=0
    while( first<=last and f==0):
        mid = int((first + last)/2)
        if list1[mid] == item :
            f = 1
            a=mid
        else:
            if item < list1[mid]:
                last = mid - 1
            else:
                first = mid + 1    
    if f==1:
        return(a)
    else:
        return("not found")
        
print(binary_search([1,5,8,3,9], 6))
print(binary_search([1,5,8,3,9], 5))

File: Searching/binarysearch.c
#include<stdio.h>
 
void main()
{
    int a[50],i,n,x,flag=0,first,last,mid;
 
    printf("Enter size of array:");
    scanf("%d",&n);
    printf("\nEnter array elements(ascending order)\n");
 
    for(i=0;i<n;++i)
        scanf("%d",&a[i]);
 
    printf("\nEnter the element to search:");
    scanf("%d",&x);
 
    first=0;
    last=n-1;
 
    while(first<=last)
    {
        mid=(first+last)/2;
 
        if(x==a[mid]){
            flag=1;
            break;
        }
        else
            if(x>a[mid])
                first=mid+1;
            else
                last=mid-1;
    }
 
    if(flag==1)
        printf("\nElement found at position %d",mid+1);
    else
        printf("\nElement not found");
 
    
}

